### PR TITLE
Revert "build: more robustly check for fcf-protection support"

### PR DESCRIPTION
**Pull request by fanquake**
We no-longer support Clang 7 (#24164). Introduced in #20720.

This reverts commit e9189a750b237eba1befc6b16c12c2cee3e0176c.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -935,10 +935,7 @@ if test "$use_hardening" != "no"; then
   AX_CHECK_COMPILE_FLAG([-Wstack-protector], [HARDENED_CXXFLAGS="$HARDENED_CXXFLAGS -Wstack-protector"])
   AX_CHECK_COMPILE_FLAG([-fstack-protector-all], [HARDENED_CXXFLAGS="$HARDENED_CXXFLAGS -fstack-protector-all"])
 
-  dnl -fcf-protection used with Clang 7 causes ld to emit warnings:
-  dnl ld: error: ... <corrupt x86 feature size: 0x8>
-  dnl Use CHECK_LINK_FLAG & --fatal-warnings to ensure we won't use the flag in this case.
-  AX_CHECK_LINK_FLAG([-fcf-protection=full], [HARDENED_CXXFLAGS="$HARDENED_CXXFLAGS -fcf-protection=full"], [], [$LDFLAG_WERROR])
+  AX_CHECK_COMPILE_FLAG([-fcf-protection=full],[HARDENED_CXXFLAGS="$HARDENED_CXXFLAGS -fcf-protection=full"])
```
Comment (hebasto) [inline configure.ac:938]: style nit: Add a space after the comma?
```suggestion
  AX_CHECK_COMPILE_FLAG([-fcf-protection=full], [HARDENED_CXXFLAGS="$HARDENED_CXXFLAGS -fcf-protection=full"])
```

**Review by hebasto — APPROVED** [review]
ACK 460854c1eba7a509327d17fd995b27d9cc8ad5f6

File: configure.ac (hunk 1 above)
Comment (fanquake) [reply configure.ac]: sure

**Review by hebasto — APPROVED** [review]
re-ACK a7973bf8f46a7fe1c70d33a1098da6aba66baaed

**Pull request merged**